mmv1/third_party/terraform/README.md

```diff
@@ -226,7 +226,134 @@ See [Tests that use a beta feature](#tests-that-use-a-beta-feature)
 
 ### Beta Feature
 
-#### Add or update a beta feature
+When the underlying API of a feature is not final (i.e. a `vN` version like
+`v1` or `v2`), is in preview, or the API has no SLO we add it to the
+`google-beta` provider rather than the `google `provider, allowing users to
+self-select for the stability level they are comfortable with.
+
+Both the `google` and `google-beta` providers operate off of a shared codebase,
+including for handwritten code. MMv1 allows us to write Go source files as
+`.go.erb` templated source, and renders them as `.go` files in the downstream
+repo.
+
+The sole generator feature you need to be aware of is a "version guard", what is
+effectively a preprocessor directive implemented using Embedded Ruby (ERB). A
+version guard is a snippet used across this codebase by convention guarding
+versioned code on an `unless` clause in a version check. For example:
+
+```
+	networkInterfaces, err := expandNetworkInterfaces(d, config)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating network interfaces: %s", err)
+	}
+<% unless version == 'ga' -%>
+	networkPerformanceConfig, err := expandNetworkPerformanceConfig(d, config)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating network performance config: %s", err)
+	}
+<% end -%>
+```
+
+In the snippet above, the `networkInterfaces` field is generally available and
+is not guarded. The `networkPerformanceConfig` field is only available at beta,
+and is guarded by `unless version == ga`, and the guarded block is terminated by
+an `end` statement.
+
+If a service includes handwritten resources and mixed features or resources at
+different versions, the client libraries used by each provider must be switched
+using guards so that the stability level of the client library matches that of
+the provider. For example, all handwritten Google Compute Engine (GCE) files
+have the following guarded import:
+
+```
+<% if version == "ga" -%>
+	"google.golang.org/api/compute/v1"
+<% else -%>
+	compute "google.golang.org/api/compute/v0.beta"
+<% end -%>
+```
+
+This is not necessary for beta-only services, or for services that are generally
+available in their entirety.
+
+#### Adding a beta resource
```

Comment:
Would you mind also changing the reference in ToC, since we change the title here?